File: 17s2/lab06/whale.pl
#!/usr/bin/perl -w

if (@ARGV != 1) {
	die "Usage: $0 <whale species>\n"
	
}
$species_src = $ARGV[0];
#print $species_src;
$total_num = 0;
$n_cnt = 0;
while ($line = <STDIN>) {
	#print $line;
	if ($line =~ /^\s*(\d+)\s*(.+)$/) {
		$num = $1;
		$content = $line;
		$content =~ s/^\ *\d+\ *//g;
		$content =~ s/\n//g;
		$species_src =~ s/\ +/\ /g;
		#print "$content\n";
		#print "$species_src\n";
		if ($content eq $species_src) {
			#print "------------------------------------------\n";
			$n_cnt++;
			$total_num+=$num;
		}
	}
}
print "$species_src observations: $n_cnt pods, $total_num individuals\n";

File: 17s2/lab06/whale.py
#!/usr/bin/python
import re, sys
if len(sys.argv) != 2:
    print("Usage: {} <whale species>\n".format(sys.argv[0]))
    sys.exit(1)
t_species = sys.argv[1]
pods = 0;idiv = 0
syntax = re.compile(r'^\s*(\d+)\s*(.+)$')
for line in sys.stdin:
	if (re.match(syntax, line)):
		#print("-----------------")
		line = line.lower()
		tr_to_single = re.compile(r's(\s*)$')
		del_space_to_single = re.compile(r'\s+')
		del_cnt = re.compile(r'^\s*(\d+)\s*')
		cnt = int(re.search(r'(\d+)\s*',line).group(0))
		line = re.sub(del_space_to_single, ' ', line)
		line = re.sub(del_cnt, '', line)
		line = re.sub(tr_to_single, '\n', line)
		species = line.strip()
    
		if species == t_species:
			pods += 1
			idiv += cnt
print("{} observations: {} pods, {} individuals".format(t_species, pods, idiv))
